frozen_lake.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+import sys
+from six import StringIO, b
+from gym import utils
+import discrete_env
+
+# Mapping between directions and index number
+LEFT = 0
+DOWN = 1
+RIGHT = 2
+UP = 3
+
+# Maps for the two different environments
+MAPS = {
+    "4x4": [
+        "SFFF",
+        "HFFG" "SFFFFFFF",
+        "FFFFFFFF",
+        "FFFHFFFF",
+        "FHFFFHFF",
+        "FFFHFFFF",
+        "FFHFFFHF",
+        "FHFFHFHF",
+        "FFFHFFFG",
+    ],
+}
+
+
+class FrozenLakeEnv(discrete_env.DiscreteEnv):
+    """
+    Winter is here. You and your friends were tossing around a frisbee at the park
+    when you made a wild throw that left the frisbee out in the middle of the lake.
+    The water is mostly frozen, but there are a few holes where the ice has melted.
+    If you step into one of those holes, you'll fall into the freezing water.
+    you navigate across the lake and retrieve the disc.
+    The surface is described using a grid like the following
+
+        SFFF
+        FHFH
+        FFFH
+        HFFG
+
+    S : starting point, safe
+    F : frozen surface, safe
+    H : hole, fall to your doom
+    G : goal, where the frisbee is located
+
+    The episode ends when you reach the goal or fall in a hole.
+    You receive a reward of 1 if you reach the goal, and zero otherwise.
+
```

Comment:
Since `nS`, `nA`, `P`, and `isd` in `super(FrozenLakeEnv, self).__init__(nS, nA, P, isd)` don't have typehints or default values, you need to add them to the docstring following the [docstring format in FastForest for classes with params](https://github.com/ThrunGroup/FastForest/blob/main/data_structures/wrappers/extremely_random_forest_classifier.py#L8-L16)